python-checks/src/main/java/org/sonar/python/checks/tests/DedicatedAssertionCheck.java

```diff
@@ -0,0 +1,239 @@
+      if (!qualifiedExpression.qualifier().is(Tree.Kind.NAME) || !((Name) qualifiedExpression.qualifier()).name().equals("self")) {
+        return;
+      }
+      String calledMethod = qualifiedExpression.name().name();
+
+      switch (calledMethod) {
+        case ASSERT_TRUE:
+          checkAssertTrueOrFalse(callExpression, ctx, true);
+          break;
+        case ASSERT_FALSE:
+          checkAssertTrueOrFalse(callExpression, ctx, false);
+          break;
+        case ASSERT_EQUAL:
+          checkAssertEqualOrNotEqual(callExpression, ctx, true);
+          break;
+        case ASSERT_NOT_EQUAL:
+          checkAssertEqualOrNotEqual(callExpression, ctx, false);
+          break;
+        case ASSERT_ALMOST_EQUAL:
+          checkAssertAlmostEqualOrNot(callExpression, ctx, ASSERT_ALMOST_EQUAL);
+          break;
+        case ASSERT_NOT_ALMOST_EQUAL:
+          checkAssertAlmostEqualOrNot(callExpression, ctx, ASSERT_NOT_ALMOST_EQUAL);
+          break;
+        default:
+          // Nothing to do
+      }
+    });
+  }
+
+  @Override
+  public CheckScope scope() {
+    return CheckScope.ALL;
+  }
+  
+  private static void checkAssertTrueOrFalse(CallExpression callExpression, SubscriptionContext ctx, boolean isAssertTrue) {
+    Map<String, String> suggestions = isAssertTrue ? ASSERT_TRUE_SUGGESTIONS : ASSERT_FALSE_SUGGESTIONS;
+    List<Argument> arguments = callExpression.arguments();
+    if (arguments.isEmpty()) {
+      return;
+    }
+    RegularArgument firstArg = TreeUtils.nthArgumentOrKeyword(0, "expr", arguments);
+    if (firstArg == null) {
+      return;
+    }
+    Expression expression = Expressions.removeParentheses(firstArg.expression());
+    String dedicatedAssertion = null;
+    if (expression.is(Tree.Kind.COMPARISON)) {
+      BinaryExpression binaryExpression = (BinaryExpression) expression;
+      if (binaryExpression.leftOperand().is(Tree.Kind.COMPARISON)) {
+        // Avoid reporting on chained comparisons (due to how comparisons are defined in the grammar, only the left operand can be a comparison)
+        return;
+      }
+      dedicatedAssertion = suggestions.get(binaryExpression.operator().value());
+    } else if (expression.is(Tree.Kind.IN)) {
+      dedicatedAssertion = suggestions.get(IN);
+    } else if (expression.is(Tree.Kind.IS)) {
+      dedicatedAssertion = ((IsExpression) expression).notToken() == null ? suggestions.get(IS) : suggestions.get(IS_NOT);
+    } else if (isCallTo(IS_INSTANCE, expression)) {
+      dedicatedAssertion = suggestions.get(IS_INSTANCE);
+    }
+    if (dedicatedAssertion != null) {
+      ctx.addIssue(callExpression, String.format(MESSAGE, dedicatedAssertion));
+    }
+  }
+
+  private static void checkAssertEqualOrNotEqual(CallExpression callExpression, SubscriptionContext ctx, boolean isAssertEqual) {
+    List<Argument> arguments = callExpression.arguments();
+    if (arguments.size() < 2) {
+      return;
+    }
```

Comment:
This check does not seems to be required since you are testing just after that both firstArg and secondArg are not null.

Reply:
Good point! Thanks!